visualizer/visualizer.go

```diff
@@ -3,24 +3,60 @@ package visualizer
 
 import (
 	"fmt"
+	"net/url"
 
+	"github.com/agneum/plan-exporter/client"
 	"github.com/agneum/plan-exporter/pgscanner"
-	"github.com/agneum/plan-exporter/visualizer/dalibo"
-	"github.com/agneum/plan-exporter/visualizer/depesz"
-	"github.com/agneum/plan-exporter/visualizer/tensor"
 )
 
-// New creates a new query plan exporter by visualizer type.
-func New(visualizer string) (pgscanner.PlanExporter, error) {
+// Visualizer consists of a URL for HTTP POST call, along with HTML element name
+type Visualizer struct {
+	postURL string
+	planKey string
+}
+
+// Export posts plan to a visualizer and returns link to the visualization plan page.
+func (v *Visualizer) Export(plan string) (string, error) {
+	formVal := url.Values{v.planKey: []string{plan}}
+
+	explainURL, err := client.MakeRequest(v.postURL, formVal)
+	if err != nil {
+		return "", fmt.Errorf("failed to make a request: %w", err)
+	}
+
+	return explainURL, nil
+}
+
+// New creates a new query plan exporter
+func New(visualizer string, url string, key string) (pgscanner.PlanExporter, error) {
 	switch visualizer {
-	case dalibo.VisualizerType:
-		return dalibo.New(), nil
+	case "custom":
+		if url == "" || key == "" {
+			return nil, fmt.Errorf("both plan_url and plan_key must be provided for custom targets")
+		}
+
+		return &Visualizer{
+			postURL: url,
+			planKey: key,
+		}, nil
+
+	case "dalibo":
+		return &Visualizer{
+			postURL: "https://explain.dalibo.com/new",
```

Comment:
Using a constant has several advantages. This makes the code more robust and modifiable. I would rather keep the constants.

Reply:
Cool--I'm actually kind of new to golang, so I'm not very familiar with the advantages/disadvantages of using constants

Reply:
FYI I have reverted the changes so the constants are back